### DISCUSSION TITLE
bug or feature? incomplete search result when scanning content

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#1857] [Q&A] [answered]
When I don't scan the content, then I get all files containing the search string

![Screenshot_2022-09-26-05-27-15-407_net.gsantner.markor.jpg](https://user-images.githubusercontent.com/9020187/192188169-2a6f4fb0-1476-4f8f-b86b-78ef111eb288.jpg)

![Screenshot_2022-09-26-05-27-55-595_net.gsantner.markor.jpg](https://user-images.githubusercontent.com/9020187/192188243-b2a66c5e-ad77-4ff0-85bb-7397805c6f58.jpg)

but when also searching the content, it doesn't show all files. no matter about the option to stop after the first result.

![Screenshot_2022-09-26-05-30-04-353_net.gsantner.markor.jpg](https://user-images.githubusercontent.com/9020187/192188441-fc1867a5-cd10-4f2d-9ab4-0a360a8ee674.jpg)

![Screenshot_2022-09-26-05-31-15-961_net.gsantner.markor.jpg](https://user-images.githubusercontent.com/9020187/192188529-4c479104-eaab-4a43-b93d-46cabd0df153.jpg)

I would say, it's a bug. Because searching also in the content should bring more results, but not less.

Maybe the issue is related to the hierarchical naming system, which I applied yesterday to most of my notes?

I use something like 
project.task.detail.andsoon.adoc

**Comment by gsantner** [comment, marked as answer]
And do really the other files also contain mieter? Search-in-content does exactly that, searching in content and displaying all matches in a file. Not searching by filename, not a combination. Thats how it works. Hierarchical: Search is recursive.

**Comment by aisbergde** [comment]
That was not clear to me that EITHER in the file name OR in the content is searched.

This then results in a feature request for another possibility:

Search in filename OR in content (logical OR).

**Comment by harshad1** [comment]
Note that with my latest changes filenames will be searched for as well when searching in content
